core/services/ocr2/plugins/dkg/persistence/db.go

```diff
@@ -0,0 +1,185 @@
+package persistence
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+	ocr2vrftypes "github.com/smartcontractkit/ocr2vrf/types"
+	"github.com/smartcontractkit/ocr2vrf/types/hash"
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+)
+
+var (
+	_        ocr2vrftypes.DKGSharePersistence = &shareDB{}
+	zeroHash hash.Hash
+)
+
+type shareDB struct {
+	q    pg.Q
+	lggr logger.Logger
+}
+
+// NewShareDB creates a new DKG share database.
+func NewShareDB(db *sqlx.DB, lggr logger.Logger, cfg pg.QConfig) ocr2vrftypes.DKGSharePersistence {
+	return &shareDB{
+		q:    pg.NewQ(db, lggr, cfg),
+		lggr: lggr,
+	}
+}
+
+// WriteShareRecords writes the provided (already encrypted)
+// share records to the Chainlink database.
+func (s *shareDB) WriteShareRecords(
+	ctx context.Context,
+	cfgDgst ocrtypes.ConfigDigest,
+	keyID [32]byte,
+	shareRecords []ocr2vrftypes.PersistentShareSetRecord,
+) error {
+	lggr := s.lggr.With(
+		"configDigest", hexutil.Encode(cfgDgst[:]),
+		"keyID", hexutil.Encode(keyID[:]))
+
+	start := time.Now()
+	defer func() {
+		lggr.Infow("Inserted DKG shares into DB", "duration", time.Since(start))
+	}()
```

Comment:
would it be helpful to have this as a prom metric also?

Reply:
Yeah, good idea!

Reply:
Added.